src/views/List/List.jsx

```diff
@@ -8,19 +8,73 @@ export default function List({ token }) {
   useEffect(() => {
     const unsubscribe = onSnapshot(collection(db, token), (snapshot) => {
       const snapshotDocs = [];
-      snapshot.forEach((doc) => snapshotDocs.push(doc.data().property));
+      snapshot.forEach((doc) => {
+        const data = doc.data();
+        data.id = doc.id;
+        snapshotDocs.push(data);
+      });
       setData(snapshotDocs);
     });
     return () => {
       unsubscribe();
     };
   }, [token]);
 
+  // on each re-render, we loop through the react data array to check the current time against the last purchased at time
+  useEffect(() => {
+    data.forEach((item) => {
+      const now = Date.now();
+      const delta = now - item.lastPurchasedAt;
+      // we then feed the delta and the item into our unCheck function and if the item has a delta greater than 86400000 we run update the isActive property in firebase
+      unCheckItem(item, delta);
+    });
+  });
+
+  const unCheckItem = async (item, delta) => {
+    if (delta > 86400000) {
+      item.isActive = false;
+      const updatedItem = doc(db, token, item.id);
+      await updateDoc(updatedItem, {
+        isActive: false,
+      });
+    }
+  };
+
+  const onChange = async (listItem) => {
+    //when an item is checked, we map through the data array and update the React state
+    const { isActive, id } = listItem;
+    const nextData = data.map((item) => {
+      if (item.id === id) {
+        item.isActive = !item.isActive;
```

Comment:
I like the use of isActive and setting that in the database, there were lots of different ways to approach this and this is direct and efficient, not using additional properties you dont need but making it really clear about the properties you are using